src/components/Auth/MAPI/MapiUnauthorized.tsx

```diff
@@ -80,6 +81,8 @@ const MapiUnauthorized: React.FC<IMapiUnauthorizedProps> = ({
       messageTTL.MEDIUM
     );
 
+    dispatch(clearImpersonationAction());
+
     reloadOrganizations();
   };
 
```

Comment:
In this component we use local state for the impersonation metadata
```
const [impersonationMetadata, setImpersonationMetadata] =
useState<IOAuth2ImpersonationMetadata | null>(null);
```

Maybe we can take it from the redux store now?